frontend/src/metabase/dashboards/components/DashboardList.jsx

```diff
@@ -12,58 +11,157 @@ import * as Urls from "metabase/lib/urls";
-              onMouseLeave={() => setHover(false)}>
+class DashboardListItem extends Component {
+    props: DashboardListItemProps
+
+    state = {
+        hover: false,
+        fadingOut: false
+    }
+
+    render() {
+        const {dashboard, setFavorited, setArchived} = this.props
+        const {hover, fadingOut} = this.state
+
+        const {id, name, created_at, archived, favorite} = dashboard
+
+        const archivalButton =
+            <Tooltip tooltip={archived ? "Unarchive" : "Archive"}>
+                <Icon
+                    className="flex cursor-pointer text-light-blue text-brand-hover ml2 archival-button"
+                    name={archived ? "unarchive" : "archive"}
+                    size={21}
+                    onClick={(e) => {
+                        e.preventDefault();
+                        this.setState({fadingOut: true})
+                        setTimeout(() => setArchived(id, !archived, true), 300);
+                    } }
+                />
+            </Tooltip>
+
+        const favoritingButton =
+            <Tooltip tooltip={favorite ? "Unfavorite" : "Favorite"}>
+                <Icon
+                    className={cx(
+                        "flex cursor-pointer ml2 favoriting-button",
+                        {"text-light-blue text-brand-hover": !favorite},
+                        {"text-gold": favorite}
+                    )}
+                    name={favorite ? "star" : "staroutline"}
+                    size={22}
+                    onClick={(e) => {
+                        e.preventDefault();
+                        setFavorited(id, !favorite)
+                    } }
+                />
+            </Tooltip>
+
+        const dashboardIcon =
             <Icon name="dashboard"
-                  className={cx("pr2", {"text-grey-1": !hover}, {"text-brand-darken": hover})} size={32}/>
-            <div className={cx("flex-full shrink-below-content-size")}>
-                <h4 className={cx("text-ellipsis text-nowrap overflow-hidden text-brand", {"text-white": hover})}
-                    style={{marginBottom: "0.2em"}}>
-                    <Ellipsified>{dashboard.name}</Ellipsified>
-                </h4>
-                <div
-                    className={cx("text-smaller text-uppercase text-bold", {"text-grey-3": !hover}, {"text-grey-2": hover})}>
-                    {/* NOTE: Could these time formats be centrally stored somewhere? */}
-                    {moment(dashboard.created_at).format('MMM D, YYYY')}
-                </div>
-            </div>
-        </Link>
-    </li>
-);
+                  className={cx("ml2", "text-grey-1")}
+                  size={25}/>
+
+        return (
+            <li className="Grid-cell shrink-below-content-size" style={{maxWidth: "550px"}}>
+                <Link to={Urls.dashboard(id)}
+                      data-metabase-event={"Navbar;Dashboards;Open Dashboard;" + id}
+                      className={cx(
```

Comment:
Don't actually need to use `cx` here since there aren't any conditional classes on this element.